design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,207 @@
+* Common Templates transition to VolumeReferences for DataVolumeTemplate source
+* SSP Operator creates VolumeImportPollers for automatic PVC import
+
+**Phase3** - HCO and UI
+* HCO Team integrates with SSP Operator’s API for setting volume import sources
+* UI Team transitions to detecting boot source using a linked VolumeReference rather than looking directly for a PVC of a specific name.
+
+
+
+# API Examples
+## VM + DataVolumeTemplate + VolumeReference as source
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+   name: my-fedora-vm-image
+spec:
+   pvc:
+     accessModes:
+     - ReadWriteOnce
+     resources:
+       requests:
+         storage: 2Gi
+   source:
+     volumeReference:
+       name: fedora
+       namespace: some-namespace
+```
+
+## VolumeImportPoller Example
+
+This example polls for image tags for the quay.io/kubevirt/fedora image and imports the latest image sorting tags based on semver. As new tags are introduced, the VolumeImportPoller will import them as new PVCs and automatically manage updating the corresponding VolumeReference.
+
+* **source** specifies where to poll from
+* **pollInterval** specifies how often to look for new imports 
+* **garbageCollectOutdated** specifies whether old PVCs should be cleaned up after a new PVC is imported.
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeImportPoller
+metadata:
+   name: fedora-poller
+spec:
+   source:
+     registry:
+       url: docker://quay.io/kubevirt/fedora
+    pollInterval: 12h
+    garbageCollectOutdated: true
+status:
+  managedVolumeReference:
+      name: fedora
+```
+
+## VolumeReference Example
+
+This example shows how a PVC is linked using the claimName field on the VolumeReference spec. The VolumeReference’s status section will indicate if the claimName is linked or dangling. 
+
+The UI can use the phase field to determine if a boot source exists for a VolumeReference or not. 
+
+* **Phase: linked** indicates that the the claimName is set and that the corresponding PVC exists and is populated
+* **Phase: dangling** indicates that the claimName points to a non-existent PVC, meaning the underlying PVC has either been deleted or is not created yet. 
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeReference
+metadata:
+   name: fedora
+spec:
+    source:
+        pvc:
+            namespace: somenamespace
+            name: fedora-32-001
+status:
+    phase: linked
+```
+
+## SSP Operator CR Example
+
+This example shows how the SSP operator could expose opting into automatic image delivery and updates for images associated with their common templates. 
+
+By default, the ssp operator would simply create dangling VolumeReferences to match the VolumeReferences used as DataVolumeTemplate sources in the common templates. Users would import their own disks which would automatically bind to the dangling VolumeReferences and make them linked to a PVC. This matches the behavior that exists today.
```

Comment:
Can you describe how a user imported pvc would automatically bind to the dangling VolumeReference?
Would this be naming related or how would the user tell the VolumeReference to reference this new PVC (without manually patching it).

Reply:
>Can you describe how a user imported pvc would automatically bind to the dangling VolumeReference?
Would this be naming related or how would the user tell the VolumeReference to reference this new PVC (without manually patching it).

yeah, i improved my wording here a bit. It was slightly confusing. Take a look at the new `Transitioning the CommonTemplates to VolumeReferences` section I added as well. It aims to clarify this specific question

To summarize, the VolumeReference can point to a source PVC that doesn't exist yet, just like the DataVolumeTemplates section in the current common templates can. When a user imports a PVC that matches the source PVC in a dangling VolumeReference, then the VolumeReference automatically gets linked.